broker/src/main/java/org/apache/rocketmq/broker/longpolling/PullRequestHoldService.java

```diff
@@ -36,6 +36,8 @@ public class PullRequestHoldService extends ServiceThread {
     private final SystemClock systemClock = new SystemClock();
     protected ConcurrentMap<String/* topic@queueId */, ManyPullRequest> pullRequestTable =
         new ConcurrentHashMap<>(1024);
+    protected ConcurrentMap<String/* topic@queueId */, Long/* offset */> lastNotifyMaxOffset =
```

Comment:
When deleting a topic or reducing the number of queues, the memory of lastNotifyMaxOffset may be leaked.

Reply:
> When deleting a topic or reducing the number of queues, the memory of lastNotifyMaxOffset may be leaked.

Indeed. But the leak is very slight? The pullRequestTable has the same issue.

Reply:
> When deleting a topic or reducing the number of queues, the memory of lastNotifyMaxOffset may be leaked.

Resolved. It is more suitable to use expired cache here. @xdkxlk PTAL.